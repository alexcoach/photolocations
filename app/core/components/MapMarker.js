// @flow

import React from 'react';
import { Marker, Callout } from 'react-native-maps';

export default class MapMarker extends React.PureComponent<*>{
    onCalloutPress = () => {
        const { onCalloutPress } = this.props;
        if (onCalloutPress) {
            onCalloutPress(this.marker);
        }
    };
    getTitle = () => {
        const { title } = this.props;
        if (title && !React.isValidElement(title)) {
            return title;
        }
        return undefined;
    };
    getRef = (ref: any) => {
        if (!this.marker) {
            this.marker = ref;
        }
    };
    marker: any;
    renderCallout = () => {
        const { title } = this.props;
        if (!title || !React.isValidElement(title)) {
            return null;
        }
        return (
            <Callout>{title}</Callout>
        );
    };
    render() {
        return (
            <Marker
                ref={this.getRef}
                {...this.props}
                title={this.getTitle()}
                onCalloutPress={this.onCalloutPress}
            >
                {this.renderCallout()}
            </Marker>
        );
    }
}
