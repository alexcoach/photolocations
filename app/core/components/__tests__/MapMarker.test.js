import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { Text } from 'native-base';
import MapMarker from '../MapMarker';

it('<MapMarker> renders correctly', () => {
    const wrapper = shallow(
        <MapMarker coordinate={{ latitude: 0, longitude: 0 }} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
});
