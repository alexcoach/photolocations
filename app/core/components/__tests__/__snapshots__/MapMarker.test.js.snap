// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapMarker> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapMarker
    coordinate={
      Object {
        "latitude": 0,
        "longitude": 0,
      }
    }
    onCalloutPress={[Function]}
    stopPropagation={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": null,
      "coordinate": Object {
        "latitude": 0,
        "longitude": 0,
      },
      "image": undefined,
      "onCalloutPress": [Function],
      "onPress": [Function],
      "stopPropagation": false,
      "style": Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ],
      "title": undefined,
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": null,
        "coordinate": Object {
          "latitude": 0,
          "longitude": 0,
        },
        "image": undefined,
        "onCalloutPress": [Function],
        "onPress": [Function],
        "stopPropagation": false,
        "style": Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
          },
          undefined,
        ],
        "title": undefined,
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
